### PR TITLE
Update Electron to v6

### PR DESCRIPTION
**Pull request by razfriman**
Update to Electron 6.0.0.

Update packages:
- `electron`
- `electron-builder`
- `electron-rebuild`

Update the `nodeIntegration` default which was changed in Electron 5.0.0.

Add missing `plist` package required from updating the packages

### CODE REVIEW

**Comment by razfriman** [comment]
Anyone able to help me figure out why the CI is failing. I don't seem to be getting all of these warnings and errors when I build the app locally

**Comment by Stanzilla** [comment]
You need to update node-pty. the old version does not work with node 10+ afaik

**Comment by Stanzilla** [comment]
Hey @razfriman are you willing to pick this up again?

**Comment by Stanzilla** [comment]
@Tyriar any idea why this fails? https://travis-ci.org/zeit/hyper/builds/583470356?utm_source=github_status&utm_medium=notification

**Comment by razfriman** [comment]
I seem to be having issues doing this upgrade actually. Things don't seem to be working well for me with node 12 and electron 6. Perhaps someone else can take this over

File: package.json [hunk 1]
```diff
@@ -219,18 +219,20 @@
...
+    "electron-rebuild": "1.8.5",
     "eslint": "4.7.2",
     "eslint-config-prettier": "2.6.0",
     "eslint-plugin-prettier": "2.3.1",
     "eslint-plugin-react": "7.3.0",
     "husky": "0.14.3",
     "inquirer": "5.1.0",
-    "node-gyp": "3.6.2",
+    "node-gyp": "5.0.3",
+    "node-pty": "^0.8.1",
```
Comment (Tyriar) [inline package.json]: @Stanzilla this is the problem, you need to get on one of the newer beta builds for Electron 6 support. Try `node-pty@0.9.0-beta26`

Comment (Stanzilla) [reply package.json]: @Tyriar hrm, still errors :(

**Comment by Tyriar** [comment]
@Stanzilla You might need to also tweak the version of node you're building with. @deepak1556 what version of node.js should builds be using with Electron 6?

**Comment by Stanzilla** [comment]
```
C:\projects\hyper\app\node_modules\node-pty>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild ) 
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
``` 
it seems to be using an old version of node-gyp, not sure why

**Comment by Stanzilla** [comment]
> @Stanzilla You might need to also tweak the version of node you're building with. @deepak1556 what version of node.js should builds be using with Electron 6?

https://electronjs.org/blog/electron-6-0 says 12.4.0, CI seems to use `gyp info using node@12.6.0 | win32 | x64` so should be fine?

**Comment by deepak1556** [comment]
@Stanzilla electron@6.0.8 will break native module compilation on windows due to https://github.com/electron/electron/issues/20185, it has been fixed yesterday and new version will be released today, please update to that when its available.

**Comment by Stanzilla** [comment]
@deepak1556 oh interesting, thanks! I hope this really is it then, because I tried with 6.0.0 before and it was the same issue

**Comment by razfriman** [comment]
For reference, I can run and build successfully on Mac OSX and fully build (`yarn run dist`) Hyper on at this point in time. I will try to see if I can replicate the appveyor failures on a Windows machine. Not sure what else to try though

**Comment by ivanwonder** [comment]
the is my pr #3781, one is to update the node-pty, another is to add a config for `experimentalUseConpty`. now this pr update the node-pty, @Tyriar I hope you can look at another problem. when open hyper with hyper cli, the node-pty exit with code 0. I test with `experimentalUseConpty` false. then the problem disappears. I found that the causes of this problem are that the node-pty create in the electron main process and hyper open by the cli. I hope it's helpful.

**Comment by raz-canva** [comment]
> Okay thanks a lot for your help @Tyriar
> 
> Now that we got this working, @razfriman I tested it a bit and panes are a bit wonky when you split them up, can you repro that on macOS?
> 
> image

Yes I am noticing the same thing in MacOS. I will try to find the cause+fix but this is my first change to Hyper, so I don't know the codebase too well

**Comment by Tyriar** [comment]
I'm not sure what that would be, other than an issue with Electron 6 and canvas? VS Code just updated and a new build will be out on Monday so I'll probably find out quick if it is an Electron 6 issue.

**Comment by Stanzilla** [comment]
@Tyriar anything came up in VSCode?

**Comment by Tyriar** [comment]
Haven't seen anything like this, related to your layout code I guess?

**Comment by Stanzilla** [comment]
@raz-canva any news? did you find something? I didn't :(

**Comment by Stanzilla** [comment]
merged the latest stuff from canary in and tried to get it working more, currently stuck because Windows thinks that `'build' is not recognized as an internal or external command,` investigating...

**Comment by Stanzilla** [comment]
next problem: 
```

 internal/modules/cjs/loader.js:630
    throw err;
    ^

Error: Cannot find module 'C:\Users\Benjamin\VCS\Git\hyper\node_modules\electron\dist\resources\electron.asar\browser\init.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:627:15)
    at Function.Module._load (internal/modules/cjs/loader.js:531:27)
    at Function.Module.runMain (internal/modules/cjs/loader.js:850:10)
    at internal/main/run_main_module.js:17:11 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
error Command failed with exit code 1.
```

**Comment by raz-canva** [comment]
I have not been able to figure out the reason for the distorted size of text when splitting the panes. 

The “build” not found error should be resolved in the package json file by replacing 
“Build” with “electron-builder” (that part should be in the PR diff)

**Comment by Stanzilla** [comment]
@raz-canva yea I did that, I am on the next issue already

File: package.json [hunk 2]
```diff
@@ -230,34 +230,34 @@
...
-    "react-deep-force-update": "2.1.1",
-    "react-dom": "16.2.1",
-    "react-redux": "5.0.7",
-    "recast": "0.13.0",
-    "redux": "3.7.2",
-    "redux-thunk": "2.2.0",
-    "reselect": "3.0.1",
-    "seamless-immutable": "7.1.3",
-    "semver": "5.5.0",
+    "pify": "4.0.1",
```
Comment (ivanwonder) [inline package.json]: this package `pify` should be moved to `app/package.json`

Comment (Stanzilla) [reply package.json]: @ivanwonder but lib/utils/paste.js uses it as well

**Comment by Stanzilla** [comment]
```
yarn run v1.19.1
$ electron target

 internal/modules/cjs/loader.js:630
    throw err;
    ^

Error: Cannot find module 'C:\Users\Benjamin\VCS\Git\hyper\node_modules\electron\dist\resources\electron.asar\browser\init.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:627:15)        
    at Function.Module._load (internal/modules/cjs/loader.js:531:27)
    at Function.Module.runMain (internal/modules/cjs/loader.js:850:10)
    at internal/main/run_main_module.js:17:11 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
error Command failed with exit code 1.
```

**Comment by LabhanshAgrawal** [comment]
@Stanzilla can you try to run [this](https://github.com/LabhanshAgrawal/hyper/tree/electron)

**Comment by Stanzilla** [comment]
@LabhanshAgrawal what's the difference? I only saw the changes for forwardRef. And nope, does not open with that change :(

**Comment by ivanwonder** [comment]
@LabhanshAgrawal try this https://github.com/ivanwonder/hyper/tree/raz/electron. I remove the initialization of rows and cols when creating the node-pty session. now the step is
`create node-pty session without rows and cols -> create xterm instance -> resize the node-pty`

**Comment by ivanwonder** [comment]
@Stanzilla https://github.com/ivanwonder/hyper/commit/730d34481e5a2e83edbbec654d4d4e425d4daae1

**Comment by Stanzilla** [comment]
@ivanwonder cherry picked those over! thanks!

**Comment by Stanzilla** [comment]
Looks and works perfectly so far!

**Comment by ivanwonder** [comment]
@Stanzilla I have removed the useless `fit()`. just test on your compute make sure work as expected.

**Comment by Stanzilla** [comment]
Yeah I saw, that commit i cherry-picked as well

**Comment by ivanwonder** [comment]
oh, missed it. wait for the feedback of @LabhanshAgrawal about the initial load resize issue.

**Comment by Stanzilla** [comment]
yup!

**Comment by LabhanshAgrawal** [comment]
The initial load resize is working. I can see it being rendered a bit small and then resizing but it's very quick and isn't really a problem according to me.
The test rendering thing is still there though.

**Comment by Stanzilla** [comment]
@LabhanshAgrawal can you make a screenshot/video of that remaining problem, please?

**Comment by ivanwonder** [comment]
I think the text rendering is the problem of fonts or browser. https://stackblitz.com/edit/typescript-dausj9?file=index.ts you can test with different fonts.

**Comment by LabhanshAgrawal** [comment]
@Stanzilla 
Screenshot 2019-10-10 at 19 22 08

**Comment by Stanzilla** [comment]
Oh yeah, I'd probably blame that on fonts

**Comment by LabhanshAgrawal** [comment]
@ivanwonder in this I haven't changed anything, the text looks moved up to me. Is this supposed to be like this?
Screenshot 2019-10-10 at 19 23 28

**Comment by LabhanshAgrawal** [comment]
Ok I tried with my terminal fonts. Maybe the issue is really with the fonts. If you're not facing any problem then you should ignore this as it might be an issue with my patched fonts.
Screenshot 2019-10-10 at 19 31 58

**Comment by ivanwonder** [comment]
Look the alphabet `j`, it's in the middle.

**Comment by ivanwonder** [comment]
if you mind this, try render type dom. it shows ok.

**Comment by LabhanshAgrawal** [comment]
it's ok, I'll look for new fonts

**Comment by ivanwonder** [comment]
if #3848 has been solved. maybe close.

**Comment by Stanzilla** [comment]
I will once we merge this one.

**Comment by LabhanshAgrawal** [comment]
@ivanwonder I tried out `DejaVu Sans Mono for Powerline` still have that gap at bottom.
I don't have any issue with 3.0.2 so it shouldn't be a font issue .

**Comment by Stanzilla** [comment]
@LabhanshAgrawal but that was before the electron update, caused by the xterm update, right? so not blocking this PR.

**Comment by LabhanshAgrawal** [comment]
Checked with canary, no font issue there also

**Comment by Stanzilla** [comment]
That does not answer my question though :D

**Comment by LabhanshAgrawal** [comment]
I noticed it in xterm update, but it was working on using canvas renderer. But it's not working with this PR even with canvas renderer.

**Comment by Stanzilla** [comment]
ok

**Comment by LabhanshAgrawal** [comment]
But I think the issue is small, so it should not block the PR, we can track it separately.

**Comment by Stanzilla** [comment]
yeah, agree

**Comment by LabhanshAgrawal** [comment]
there's nothing else blocking this, right?

**Comment by Stanzilla** [comment]
nope I just some last testing and everything looks good, so, merging!

**Comment by ivanwonder** [comment]
because there are different version of chrome between them , run that page on hyper 3.0.2

________________________________
发件人: Benjamin Staneck <notifications@github.com>
发送时间: Friday, October 11, 2019 3:20:47 AM
收件人: zeit/hyper <hyper@noreply.github.com>
抄送: ivan <ivanwonder@outlook.com>; Mention <mention@noreply.github.com>
主题: Re: [zeit/hyper] Update Electron to v6 (#3785)


Merged #3785<https://github.com/zeit/hyper/pull/3785> into canary.

―

**Pull request merged**